src/splinterdb.c

```diff
@@ -812,15 +852,20 @@ splinterdb_iterator_init(const splinterdb     *kvs,      // IN
    trunk_range_iterator *range_itor = &(it->sri);
 
    char start_key_buffer[MAX_KEY_SIZE] = {0};
-   if (!slice_is_null(start_key)) {
+   bool start_key_is_null              = slice_is_null(start_key);
+   if (!start_key_is_null) {
       int rc = encode_key(MAX_KEY_SIZE, start_key_buffer, start_key);
       if (rc != 0) {
          return rc;
       }
    }
 
-   platform_status rc = trunk_range_iterator_init(
-      kvs->spl, range_itor, start_key_buffer, NULL, UINT64_MAX);
+   platform_status rc =
+      trunk_range_iterator_init(kvs->spl,
+                                range_itor,
+                                (start_key_is_null ? NULL : start_key_buffer),
```

Comment:
This is the real fix.

Btw, I simply could not reproduce this problem or show any symptoms of its effect in unit-tests when run in optimized build modes.

On L 854 above, `start_key_buffer[]` would be null-terminated at byte-0, so nothing offensive seems to happen in opt-builds. The bug could only be shown to manifest in debug build modes.

That's not very comforting, as now we don't know how bad the exposure might have been prior to this bug fix.

Reply:
> On L 854 above, start_key_buffer[] would be null-terminated at byte-0, so nothing offensive seems to happen in opt-builds.

Keys are not C strings.  Keys are arbitrary sequences of bytes.  Their structure is application defined.